osquery/bro/BrokerManager.h

```diff
@@ -0,0 +1,92 @@
+#pragma once
+
+#include <broker/broker.hh>
+#include <broker/endpoint.hh>
+#include <broker/message_queue.hh>
+
+#include "osquery/bro/QueryManager.h"
+
+#include <osquery/database.h>
+#include <osquery/status.h>
+#include <osquery/system.h>
+
+#include <algorithm>
+#include <iostream>
+#include <list>
+
+namespace osquery {
+
+class BrokerManager {
+ private:
+  BrokerManager();
+
+ public:
+  // Get a singleton instance
+  static BrokerManager* getInstance() {
```

Comment:
Our existing use is the Meyer's Singleton pattern, the recommended for C++11+ codebases. This works in our multi-threaded environment too.
> `static` storage duration for a function local means that only one instance of that local exists in the program

http://stackoverflow.com/questions/17712001/how-is-meyers-implementation-of-a-singleton-actually-a-singleton